velox/common/memory/MemoryPool.h

```diff
@@ -243,6 +255,52 @@ class MemoryPool : public std::enable_shared_from_this<MemoryPool> {
     VELOX_NYI("release() needs to be implemented in derived memory pool.");
   }
 
+  /// Memory arbitration related interfaces.
+
+  /// Returns the number of bytes that haven't been reserved for use, and can be
+  /// freed by shrinking this memory pool's limit.
+  virtual uint64_t freeBytes() const = 0;
+
+  /// Invoked to free up to the specified amount of memory by shrinking this
+  /// memory pool's limit. The function returns the actually freed memory bytes.
+  /// If 'targetBytes' is zero, the function free up all the unreserved memory
+  /// bytes.
+  virtual uint64_t shrink(uint64_t targetBytes = 0) = 0;
+
+  /// Invoked to bump up the memory pool's limit by 'bytes'. The function
+  /// returns the memory pool's limit after the grow.
+  virtual uint64_t grow(int64_t bytes) = 0;
+
+  /// Returns the memory reclaimer of this memory pool if not null.
+  MemoryReclaimer* reclaimer() const;
+
+  /// Invoked by the memory arbitrator to enter memory arbitration processing.
+  /// It is a noop if 'reclaimer_' is not set, otherwise invoke the reclaimer's
```

Comment:
This is provides a simple wrapper that can check if the reclaimer is null or not? Otherwise we have to check at the caller site. Also this makes the reclaim part of a memory pool's function? WDYT?